### PR TITLE
Add integration tests

### PR DESCRIPTION
**Pull request by spbnick**
This adds integration tests, makes sure coverage data is collected from them, and makes fixes necessary for them to work.

### CODE REVIEW

File: kpet/run.py [hunk 1]
```diff
@@ -127,7 +113,8 @@ def __get_hosts(database, src_path_set):
                         case.host_type_regex or \
                         suite.host_type_regex or \
                         database.host_type_regex
```
Comment (wizardofoz123) [inline kpet/run.py:115]: I can't follow all the logic of the code, but I just need to ask: what's the difference between database.host_type_regex and host_type_regex?

File: tests/assets/suites/default/index.yaml [hunk 2]
```diff
@@ -5,8 +5,3 @@ patterns:
 cases:
     -   name: default/ltplite
         tasks: suites/default/ltplite.xml
-maintainers:
```
Comment (wizardofoz123) [inline tests/assets/suites/default/index.yaml:8]: Is this a part of the patch?

Comment (spbnick) [reply tests/assets/suites/default/index.yaml:8]: Kind-of. I'm stopping accepting unknown nodes with this patch and removing nodes I forgot to remove previously, which would prevent validation now. Should've added that to the commit description.

File: kpet/run.py (hunk 1 above)
Comment (spbnick) [reply kpet/run.py:115]: `database.host_type_regex` is a host type regex assigned at the DB-level (global default), in the database. The `host_type_regex` is the host type regex in effect for the particular test case we're iterating through, which takes into account host type regex'es at the database level (`database.host_type_regex`), suite level (`suite.host_type_regex`) and case level (`case.host_type_regex`), which override each other in this order, if defined.

**Review by wizardofoz123 — APPROVED** [review]

**Comment by spbnick** [comment]
@cki-bot, could you please test this?

**Comment by spbnick** [comment]
All pipelines that could reached Beaker. A bunch of jobs succeeded. I also checked that baseline outputs for all trees and architectures match the previous output. Merging. Thank you for the review, Jakub!

**Pull request merged**